### PR TITLE
Add notes for lecture 10: Software Testing

### PR DESCRIPTION
**Pull request by CharlesReinhardt**
This pull request adds notes for lecture 10 with a subject of software testing. 

This was our first class day investigating the software testing unit of material. It involves 
information on different types of testing, why we might use them, and particularly starts to 
focus on unit testing.

### CODE REVIEW

**Comment by CharlesReinhardt** [comment]
Per #95, I am going to try to revise this pull request to move lecture10 notes into our LectureNotes subdirectory.

**Comment by kevinaangstadt** [comment]
@pamaph20 please review this PR in the comments

**Comment by pamaph20** [comment]
Can it be assumed that the last revision commit is in fact the most recent and last change of the lecture notes??

File: LectureNotes/lecture10.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Lecture 10: Software Testing
```
Comment (pamaph20) [inline LectureNotes/lecture10.md:1]: Please review PR #116 for styling issues labeled.

File: LectureNotes/lecture10.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+If we were to try and create tests for a list in python, what might we test?
+- Does it append correctly?
+    - after appending, does the rest of the list stay the same?
+- Does it "pop" correctly?
+    - after popping, does the length of the list change?
+    - after popping, what happens when we try to access the most recently removed iten?
+- This can get complicated really quick
+
+# Software Testing
+__Def__: an investigation conducted to provide stakeholders with information about the quality of the software produt or service under review.
```
Comment (pamaph20) [inline LectureNotes/lecture10.md:18]: As per style guide #116 , definitions should be labeled 
>#### Definition
>#### Words

File: LectureNotes/lecture10.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    - Provides special functions/methods to return a boolean or certain failure assertions
+
+- Test case **discoverer**: finds all these unit tests in the code (based on a special naming scheme)
+- Test case **runner**: chooses which tests to execute
+- **Test case**: perform the following in isolation:
+    1. Establish some precondition
+    2. Perform an operation
+    3. Assert postconditions
+- **Test fixture**: surrounds a test case to provide code that is run before and after each test case
+    - often used for set up (before) and clean up (after)
```
Comment (pamaph20) [inline LectureNotes/lecture10.md:52]: As per style guide #116, must include a finishing "Something Fun" portion.

File: LectureNotes/lecture10.md (hunk 1 above)
Comment (pamaph20) [inline LectureNotes/lecture10.md:1]: In general please stay consistent in the use of punctuation.

**Comment by pamaph20** [comment]
Overall very nice pr. I appreciated the well document commit messages and the multiple small commits. General comments are simple formatting errors (see #116 ) and grammar mistakes. But overall well done, excited to see the next commits with fixes!

**Comment by kevinaangstadt** [comment]
Please make the requested changes and then I will accept this PR :-)

**Comment by CharlesReinhardt** [comment]
Revised day 10 notes. Would love if @pamaph20 could take another look at this PR before merging

**Comment by pamaph20** [comment]
Just one tiny last edit that would make it great. Make sure all sentences or statements start off with a capital and end with punctuation, this is very inconsistent throughout. Other than that the revisions look great!

File: Day10Notes.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+- "I swear I saw this and fixed this before"
+- When a developer fixes a bug, they can add a regression test that exposes this bug to make sure it stays fixed.
+
+---
+
+## System Testing
+
+> #### Definition
+> Testing an end-to-end system for correctness is called __system testing__
```
Comment (pamaph20) [inline Day10Notes.md:61]: For example end this off with a period.

File: Day10Notes.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+---
+
+## Unit Testing
+
+> #### Definition
+> testing individual chunks of source code to determine that they work is called __unit testing__
```
Comment (pamaph20) [inline Day10Notes.md:68]: And capitalize testing and add a period at the end.

**Comment by CharlesReinhardt** [comment]
Fixed grammatical errors with capitalization and punctuation (at least all the ones I found). Would love for @pamaph20 to take another look at this and point out any remaining errors I might have missed.

**Pull request merged**